src/d3a/d3a_core/user_profile_handler.py

```diff
@@ -0,0 +1,253 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import os
+from datetime import datetime
+from typing import Dict
+
+import d3a.constants
+import pytz
+from d3a.d3a_core.util import should_read_profile_from_db
+from d3a_interface.constants_limits import GlobalConfig
+from d3a_interface.constants_limits import TIME_ZONE
+from d3a_interface.read_user_profile import read_arbitrary_profile, InputProfileTypes
+from d3a_interface.utils import generate_market_slot_list
+from pendulum import DateTime
+from pendulum import instance
+from pony.orm import Database, Required, db_session, select
+from pony.orm.core import Query
+
+
+class ProfileDBConnectionHandler:
+    """
+    Handles connection and interaction with the user-profiles postgres DB via pony ORM
+    """
+    _db = Database()
+
+    class ProfileTimeSeries(_db.Entity):
```

Comment:
Why don't we set the profile time series as JSON field in the ConfigurationAreaProfileUUIDs instead of creating 2 models and instances for all rows?

Reply:
We need to be able to filter for the timestamp. And this needs to be done fast. If you have to deserialise a JSON string first and then filter for the time, you first load the entire profile into memory (which is the reason for this story to be implemented: reducing the used memory) and second it needs too much time. 
We actually tried out some DB types that are dedicated (e.g. influxDB) but Spyros found out that using a simple postgres DB an storing the profiles in a timestamp-value manner is much more performant. 
This needs to be evaluated again once we have thousands of profiles.